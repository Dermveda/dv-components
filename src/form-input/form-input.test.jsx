import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FormInput from './form-input.jsx';
const props = {
	name: 'test',
	type: 'email',
	onChange: jest.fn()
};

describe('<FormInput/>', () => {
	it('Will render', () => {
		const el = renderer.create(<FormInput {...props} />);
		expect(el).toMatchSnapshot();
	});
	it('Will be password', () => {
		const el = shallow(<FormInput {...props} type="password" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be text', () => {
		const el = shallow(<FormInput {...props} type="text" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be number', () => {
		const el = shallow(<FormInput {...props} type="number" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be date', () => {
		const el = shallow(<FormInput {...props} type="date" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be invalid type', () => {
		const el = shallow(<FormInput {...props} type="" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be invalid type', () => {
		const el = shallow(<FormInput {...props} type="" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be required', () => {
		const el = shallow(<FormInput {...props} required={true} />);
		expect(el).toMatchSnapshot();
	});
	it('Will have label', () => {
		const el = shallow(<FormInput {...props} label="test label" />);
		expect(el).toMatchSnapshot();
	});
	it('Will have help text', () => {
		const el = shallow(<FormInput {...props} helpText="help me test" />);
		expect(el).toMatchSnapshot();
	});
	it('Will have label', () => {
		const el = shallow(<FormInput {...props} label="test label" />);
		expect(el).toMatchSnapshot();
	});
	it('Will have min number', () => {
		const el = shallow(<FormInput {...props} min={1} type="number" />);
		expect(el).toMatchSnapshot();
	});
	it('Will have max number', () => {
		const el = shallow(<FormInput {...props} max={1} type="number" />);
		expect(el).toMatchSnapshot();
	});
	it('Will have placeholder', () => {
		const el = shallow(<FormInput {...props} placeholder="test ph" />);
		expect(el).toMatchSnapshot();
	});
	it('Will be uppercase', () => {
		const el = shallow(<FormInput {...props} isUppercase={true} />);
		expect(el).toMatchSnapshot();
	});
	it('Will have min length', () => {
		const el = shallow(<FormInput {...props} minLength={5} />);
		expect(el).toMatchSnapshot();
	});
	it('Will run onChange', () => {
		const onChangeFunc = jest.fn();
		const el = mount(<FormInput {...props} onChange={onChangeFunc} />);
		el.find('input').simulate('change', { target: { value: 'some value' } });
		expect(onChangeFunc).toHaveBeenCalledTimes(1);
	});
	it('Will be disabled', () => {
		const el = shallow(<FormInput {...props} disabled={true} />);
		expect(el).toMatchSnapshot();
	});
});
