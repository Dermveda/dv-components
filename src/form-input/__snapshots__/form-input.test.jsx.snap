// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput/> Will be date 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="date"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid date.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be disabled 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    disabled={true}
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be invalid type 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type=""
  />
  <div
    className="invalid-feedback"
  >
    Invalid input.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be invalid type 2`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type=""
  />
  <div
    className="invalid-feedback"
  >
    Invalid input.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be number 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="number"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be password 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="password"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a password.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be required 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  >
    *
  </label>
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    required={true}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be text 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="text"
  />
  <div
    className="invalid-feedback"
  >
    Please fill in field.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will be uppercase 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control dvc-form-group--caps"
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have help text 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  >
    help me test
  </small>
</div>
`;

exports[`<FormInput/> Will have label 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  >
    test label
  </label>
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have label 2`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  >
    test label
  </label>
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have max number 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="number"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have min length 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have min number 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    type="number"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will have placeholder 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  />
  <input
    autoComplete="off"
    className="form-control "
    id="test"
    name="test"
    onChange={[Function]}
    placeholder="test ph"
    type="email"
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;

exports[`<FormInput/> Will render 1`] = `
<div
  className="form-group dvc-form-group"
>
  <label
    htmlFor="test"
  >
    
  </label>
  <input
    autoComplete="off"
    className="form-control "
    disabled={undefined}
    id="test"
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    required={undefined}
    type="email"
    value={undefined}
  />
  <div
    className="invalid-feedback"
  >
    Please enter a valid email.
  </div>
  <small
    className="form-text text-muted"
  />
</div>
`;
